public/app/plugins/panel/graph2/GraphPanelController.tsx

```diff
@@ -113,6 +116,13 @@ export class GraphPanelController extends React.Component<GraphPanelControllerPr
     });
   }
 
+  onHorizontalRegionSelected(from: number, to: number) {
+    getTimeSrv().setTime({
```

Comment:
This feels really dirty, is there a better way that is uncoupled from the Angular service?

Reply:
yea, my thinking is to have a onChangeTimeRange on the PanelProps instead so panel plugins do not need to access TimeSrv

Reply:
Moved it to PanelChrome instead